editor-extensions/vscode/server/src/server.ts

```diff
@@ -249,7 +250,8 @@ function getDocumentSettings(resource: string): Thenable<GrainSettings> {
     result = connection.workspace.getConfiguration({
       scopeUri: resource,
       section: "grain_language_server",
-    });
+    })
+    .then(settings => settings ?? globalSettings);
```

Comment:
I'm not sure sure if this is correct because I don't understand if `scopeUri` changes the shape of what the promise resolves to, but if it's supposed to have the same shape as above then it's ok.